cmd/nodepool/aws/create.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/nodepool/core"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AWSPlatformCreateOptions struct {
+	InstanceProfile string
+	SubnetID        string
+	SecurityGroupID string
+	InstanceType    string
+	RootVolumeType  string
+	RootVolumeIOPS  int64
+	RootVolumeSize  int64
+}
+
+func NewCreateCommand(coreOpts core.CreateNodePoolOptions) *cobra.Command {
+	platformOpts := AWSPlatformCreateOptions{
+		InstanceType:   "m4.large",
+		RootVolumeType: "gp2",
+		RootVolumeSize: 120,
+		RootVolumeIOPS: 0,
+	}
+	cmd := &cobra.Command{
+		Use:          "aws",
+		Short:        "Creates basic functional NodePool resources for AWS platform",
+		SilenceUsage: true,
+	}
+
+	cmd.Flags().StringVar(&platformOpts.InstanceType, "instance-type", platformOpts.InstanceType, "The AWS instance type of the NodePool")
+	cmd.Flags().StringVar(&platformOpts.SubnetID, "subnet-id", platformOpts.SubnetID, "The AWS subnet ID in which to create the NodePool")
+	cmd.Flags().StringVar(&platformOpts.SecurityGroupID, "securitygroup-id", platformOpts.SecurityGroupID, "The AWS security group in which to create the NodePool")
+	cmd.Flags().StringVar(&platformOpts.InstanceProfile, "instance-profile", platformOpts.InstanceProfile, "The AWS instance profile for the NodePool")
+	cmd.Flags().StringVar(&platformOpts.RootVolumeType, "root-volume-type", platformOpts.RootVolumeType, "The type of the root volume (e.g. gp2, io1) for machines in the NodePool")
+	cmd.Flags().Int64Var(&platformOpts.RootVolumeIOPS, "root-volume-iops", platformOpts.RootVolumeIOPS, "The iops of the root volume when specifying type:io1 for machines in the NodePool")
+	cmd.Flags().Int64Var(&platformOpts.RootVolumeSize, "root-volume-size", platformOpts.RootVolumeSize, "The size of the root volume (min: 8) for machines in the NodePool")
+
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(context.Background())
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT)
+		go func() {
+			<-sigs
+			cancel()
+		}()
+
+		if err := CreateNodePool(ctx, coreOpts, platformOpts); err != nil {
+			log.Error(err, "Failed to create nodepool")
+			os.Exit(1)
+		}
+	}
+
+	return cmd
+}
+
+func CreateNodePool(ctx context.Context, coreOpts core.CreateNodePoolOptions, platformOpts AWSPlatformCreateOptions) error {
+	return coreOpts.CreateNodePool(ctx, platformOpts)
+}
+
```

Comment:
can we add the lines to validate each platform type implements the interface?

Reply:
Can you please elaborate on this? how do you suggest to add this validation?
Not all platform types implement the interface (Agent and None weren't added to the NodePool CLI)

Reply:
Any platform implementing PlatformOptions i.e within cmd/nodepool/* needs to validate it satisfy the interface a build time, e.g https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/internal/platform/platform.go#L19-L22

Reply:
Ok, I will add this
But why do we need it, as already we have the assigment of the implementation on the interface, which fail on build time if the implementation doesn't satisfy the interface
In this case, the call to the function this comment points to would fail in build time